### ISSUE TITLE
Dendron 0.107 Not Recognizing Journal Notes

### ISSUE THREAD

**Issue by micharris42** [dendronhq/dendron#3364]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [ ] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [X] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

After updating Dendron to version 0.107, I have received multiple notifications to the following effect: 

"Failed to read `journal.2015.06.30` in vault: Invalid regular expression: /\bC++\b/: Nothing to repeat"

This only seems to be affecting Journal notes, but there are too many to list. 

These notes do not show up when using Lookup.

## To Reproduce

1. Start VSCodium
2. Dendron initializes
3. Numerous notifications appear for journal notes with the text "Failed to read `note title` in vault: Invalid regular expression: /\bC++\b/: Nothing to repeat"
4. Notes for which there are notifications do not show up when using Lookup

## Expected behavior

That Dendron recognize these notes, as it did prior to version 0.107.

## Desktop (please complete the following information)

- OS: Windows 10
- Browser (if with published websites; include version): NA
- Versions (all that apply to bug): 
  - Dendron Extension: Dendron 0.107
  - VS Code version: VSCodium 1.70.0
  - Dendron CLI:

**Comment by natesubra** [comment]
Same issue, running VSCode (MS User Install) 1.70.0 on Win11 (21H2)

Rolling back to Dendron v0.106.0 resolves the issue. Definitely looks like something isn't happy parsing something in the note structure. All other notes appear fine. This breaks the display of the `Daily` section in the tree view. Notes are of course intact as this is just a parsing issue.

**Comment by jaydee829** [comment]
Same issue running VSCode 1.70.0 on Ubuntu 18.04. Cannot create new notes either. Rolling back to v0.106.0 resolves the issue

**Comment by hkievet** [comment]
TIL:  You can change the version of an Extension by right clicking on it and selecting "Install Another Version..." and selecting another version.

**Comment by kevinslin** [comment]
investigating

**Comment by kevinslin** [comment]
@micharris42 
do you have a copy of your dendron.yml that you can share?

**Comment by micharris42** [comment]
@kevinslin, please see below:

```
dev:
    enablePreviewV2: true
version: 5
commands:
    lookup:
        note:
            selectionMode: extract
            confirmVaultOnCreate: false
            leaveTrace: false
            bubbleUpCreateNew: false
            vaultSelectionModeOnCreate: smart
            fuzzThreshold: 0.2
    randomNote: {}
    insertNote:
        initialValue: templates
    insertNoteLink:
        aliasMode: none
        enableMultiSelect: false
    insertNoteIndex:
        enableMarker: false
    copyNoteLink: {}
    templateHierarchy: template
workspace:
    dendronVersion: 0.93.0
    vaults:
        -
            fsPath: vault
    journal:
        dailyDomain: journal
        name: journal
        dateFormat: y.MM.dd
        addBehavior: asOwnDomain
    scratch:
        name: scratch
        dateFormat: y.MM.dd.HHmmss
        addBehavior: asOwnDomain
    graph:
        zoomSpeed: 1
        createStub: false
    enableAutoCreateOnDefinition: false
    enableXVaultWikiLink: false
    enableRemoteVaultInit: true
    workspaceVaultSyncMode: noCommit
    enableAutoFoldFrontmatter: false
    maxPreviewsCached: 10
    maxNoteLength: 204800
    enableUserTags: false
    enableHashTags: true
    task:
        name: ''
        dateFormat: ''
        addBehavior: childOfCurrent
        statusSymbols:
            '': ' '
            wip: w
            done: x
            assigned: a
            moved: m
            blocked: b
            delegated: l
            dropped: d
            pending: 'y'
        prioritySymbols:
            H: high
            M: medium
            L: low
        todoIntegration: false
        createTaskSelectionType: selection2link
        taskCompleteStatus:
            - done
            - x
    enableEditorDecorations: true
    enableFullHierarchyNoteTitle: false
    templateHierarchy: template
    enableSmartRefs: false
preview:
    enableFMTitle: true
    enableNoteTitleForLink: true
    enableMermaid: true
    enablePrettyRefs: true
    enableKatex: true
    automaticallyShowPreview: false
    enableFrontmatterTags: true
    enableHashesForFMTags: false
publishing:
    enableFMTitle: true
    enableNoteTitleForLink: true
    enableMermaid: true
    enablePrettyRefs: true
    enableKatex: true
    copyAssets: true
    siteHierarchies:
        - root
    enableSiteLastModified: true
    siteRootDir: docs
    enableFrontmatterTags: true
    enableHashesForFMTags: false
    enableRandomlyColoredTags: true
    writeStubs: false
    seo:
        title: Dendron
        description: Personal knowledge space
    github:
        enableEditLink: true
        editLinkText: Edit this page on GitHub
        editBranch: main
        editViewMode: tree
    enablePrettyLinks: true
    enableTaskNotes: true
```

**Comment by natesubra** [comment]
[my dendron.yml](https://gist.github.com/natesubra/8e6985e08f1216d4bb2f3368e050eae0)

Just for variety

**Comment by suorcd** [comment]
Having a similar issue on a M1 MacBook.

**Comment by kevinslin** [comment]
heads up that we've found the issue and have released a fix in `0.107.1` (should be live in a few minutes - you might need to reload vscode to see the updated extension)

**Comment by micharris42** [comment]
@kevinslin, I have just updated to Dendron 0.107.1, and the notifications are no longer popping up and Lookup is functioning once again.

However, after the upgrade, Dendron gave a notification that my dendron.yml was missing a configuration. I went ahead with the doctor, and the `enableHandlebarTemplates` option has been re-enabled, despite being deprecated in version 0.107.

Was this intentional or in error?
